src/store/validation/index.ts

```diff
@@ -140,8 +160,11 @@ export function createDefaultJsonSchemaValidatorFactoryFor(jsonSchema: JSONSchem
   if (subSchema === '') {
     return ajv.compile(jsonSchema)
   } else {
-    ajv.addSchema(jsonSchema)
-    const validate = ajv.getSchema(subSchema)
+    let validate = ajv.getSchema(subSchema)
+    if(!validate) {
```

Comment:
@jcbashdown , does this invalid scenario occur because the `'#/definitions/IContainer'` is not considered as subSchema for `flowSpecJsonSchema.json` ?
from this code: `validators.set(validationType, createDefaultJsonSchemaValidatorFactoryFor(flowContainerJsonSchema, '#/definitions/IContainer'))`

If yes, what if we do not provide a subSchema like we did with block https://github.com/FLOIP/flow-builder/blob/a6da072d6f61e7be654ba9961f4c5fda616f1f12/src/store/validation/index.ts#L127 ?
If this will work, I suggest we remove this new condition.

Reply:
This avoids an error on the next page `Uncaught (in promise) Error: schema with key or id "" already exists`. I'm not sure it's ideal but I'm struggling to find a better way to do it... Your suggested solution still has this error. 

I think basically it's caused by calling `createDefaultJsonSchemaValidatorFactoryFor` twice with the same json schema - once for individual flows and once for the container. The only thing we do to prevent this is to call `validators.has(validationType)` but this doesn't record that all the validators in https://github.com/FLOIP/flow-runner/blob/master/dist/resources/flowSpecJsonSchema.json are already registered, only that `flow` is. That means that as long as subSchema is not a blank string we add the jsonSchema again each time we hit `ajv.addSchema(jsonSchema)` with the stuff in that file. By checking if there is something already registered for the subschema, we prevent adding the same json twice (as necessarily the sub schema would have been added already if all that json was added).

I don't think doing the check here - as close to the place we add the potentially duplicate schema as possible - is terrible but it does make me wonder about the point of `validators.has()`

Reply:
I see, thanks for raising this scenario
> calling createDefaultJsonSchemaValidatorFactoryFor twice with the same json schema
I will keep that in mind

Reply:
Very weird. When I tested this before, removing subSchema from the call didn't fix it but just been debugging another issue and making this change fixes that AND the original issue doesn't come back. Suspecting that I didn't refresh the browser the first time or something dumb like that!